app/src/main/java/com/owncloud/android/operations/RemoveFileOperation.kt

```diff
@@ -0,0 +1,98 @@
+import com.owncloud.android.lib.common.operations.RemoteOperation
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import com.owncloud.android.lib.resources.files.RemoveFileRemoteOperation
+import com.owncloud.android.operations.common.SyncOperation
+import com.owncloud.android.utils.MimeTypeUtil
+
+/**
+ * Remote operation to remove a remote file or folder from an ownCloud server.
+ *
+ * @param file  OCFile instance representing the remote file or folder to remove.
+ * @param onlyLocalCopy  If true, only the local copy will be removed (if it exists).
+ * @param user  User account associated with the operation.
+ * @param isInBackground  Flag indicating if the operation runs in the background.
+ * @param context  Android context.
+ * @param storageManager  Storage manager handling local file operations.
+ */
+class RemoveFileOperation(
+    val file: OCFile,
+    private val onlyLocalCopy: Boolean,
+    private val user: User,
+    val isInBackground: Boolean,
+    private val context: Context,
+    storageManager: FileDataStorageManager
+) : SyncOperation(storageManager) {
+
+    /**
+     * Executes the remove operation.
+     *
+     * If the file is an image, it will also be removed from the thumbnail cache.
+     * Handles both encrypted and non-encrypted files. Removes the file locally if needed.
+     *
+     * @param client  OwnCloudClient used to communicate with the remote server.
+     * @return RemoteOperationResult indicating success or failure of the operation.
+     */
+    override fun run(client: OwnCloudClient?): RemoteOperationResult<*> {
+        var result: RemoteOperationResult<*>? = null
+        val operation: RemoteOperation<*>?
+
+        // Remove image from thumbnail cache
+        if (MimeTypeUtil.isImage(file.mimeType)) {
+            ThumbnailsCacheManager.removeFromCache(file)
+        }
+
+        var localRemovalFailed = false
+
+        if (!onlyLocalCopy) {
+            operation = if (file.isEncrypted) {
+                val parent = storageManager.getFileById(file.parentId)
+                if (parent == null) {
+                    return RemoteOperationResult<Any?>(RemoteOperationResult.ResultCode.LOCAL_FILE_NOT_FOUND)
+                }
+                RemoveRemoteEncryptedFileOperation(
+                    file.remotePath,
+                    user,
+                    context,
+                    file.getEncryptedFileName(),
+                    parent,
+                    file.isFolder
+                )
+            } else {
+                RemoveFileRemoteOperation(file.remotePath)
+            }
+
+            result = operation.execute(client)
+            if (result.isSuccess || result.code == RemoteOperationResult.ResultCode.FILE_NOT_FOUND) {
+                localRemovalFailed = !storageManager.removeFile(file, true, true)
+            }
+        } else {
+            localRemovalFailed = !storageManager.removeFile(file, false, true)
```

Comment:
re-add local generation of resized images if image is deleted only locally, to save server request

```suggestion
          if (MimeTypeUtil.isImage(fileToRemove.getMimeType())) {
            // store resized image
            ThumbnailsCacheManager.generateResizedImage(fileToRemove);
        }
            localRemovalFailed = !storageManager.removeFile(file, false, true)
```